app/models/track.rb

```diff
@@ -33,6 +33,11 @@ def self.for!(param)
     find_by!(slug: param)
   end
 
+  def self.for_repo(repo)
+    slug = repo.gsub(/-(test-runner|analyzer|representer)$/, '').split('/').last
+    find_by(slug:)
+  end
```

Comment:
I've extracted this code into a helper method as it was now being used in multiple places.